software/firmware/experimental/knobs.py

```diff
@@ -334,3 +334,23 @@ def build(self) -> "KnobBank":
     @staticmethod
     def builder(knob: Knob) -> Builder:
         return KnobBank.Builder(knob)
+
+
+class BufferedKnob(Knob):
+    """A knob whose value remains fixed until .update(...) is called
+
+    This allows multiple uses of .percent(), .choice(...), etc... without forcing a re-read of
+    the ADC value
```

Comment:
[optional] I think it'd be helpful to add an example of how to create an instance of a BufferedKnob in the docstring.

Reply:
Done.